azurelinuxagent/ga/update.py

```diff
@@ -728,6 +730,36 @@ def _write_pid_file(self):
 
         return pid_files, pid_file
 
+    @staticmethod
+    def _add_sym_link_for_ub20_04():
+        # Function to create a symlink for the python interpreter that the agent is running by
+        agent_dir_python_symlink_path = get_python_symlink_path_if_ubuntu_20_04_image()
+        if agent_dir_python_symlink_path is None:
+            # Not an Ubuntu 20.04 machine, skipping creating a symlink
+            return
+
+        add_event(
+            AGENT_NAME,
+            op=WALAEventOperation.Ubuntu2004_Image,
+            version=CURRENT_VERSION,
+            is_success=True,
+            message="Agent running on Ubuntu20.04, setting a python symlink in agent directory")
+
+        if os.path.islink(agent_dir_python_symlink_path):
+            logger.info("Symlink to python interpreter already exists in agent dir {0}".format(agent_dir_python_symlink_path))
+            return
+
+        try:
+            # Create a symlink in agent directory which points to the Python interpreter path that the agent is running on
+            os.symlink(sys.executable, agent_dir_python_symlink_path)
+        except OSError as e:
```

Comment:
does symlink *always* throw an OSError?

Reply:
I couldn't find the exact list in the official documentation but read in a blog it reports FilenotFoundError, or FileExistsError as OSError. I can do a blanket catch too just for piece of mind I guess